### ISSUE TITLE
Adding new time card ends up in database error

### ISSUE THREAD

**Issue by kraj** [ledgersmb/LedgerSMB#636]
I migrated 1.3 to 1.4.10.1 and I am seeing this 

Error!

Internal Database Error
More information has been reported in the error logs at LedgerSMB.pm line 776.

dbversion: 1.4.10, company: foo

When Timecards -> Add Timecard -> Project

I have added  Projects using 1.3 as well as 1.4, it fails consistently

Reporting units -> Projects -> Add unit

**Comment by ehuelsmann** [comment]
Thanks for reporting! Your report says:

> Internal Database Error
> More information has been reported in the error logs at LedgerSMB.pm line
> 776.
>
 Could you check the logs (web server logs, usually) to see if there's more
information available there regarding the error condition?

Regards,

Erik.

**Comment by kraj** [comment]
left join account_link l
	    ON (c.id = l.account_id)
	group by c.id, c.accno, c.description, c.category, c.heading,
	         c.gifi_accno, c.contra, c.tax;
2015-04-20 02:41:29 PDT ERROR:  function public.inventory__get_item_by_partnumber(unknown) does not exist at character 15
2015-04-20 02:41:29 PDT HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
2015-04-20 02:41:29 PDT STATEMENT:  SELECT * FROM "public"."inventory__get_item_by_partnumber"(NULL)
root@ledgersmb:~# tail /var/log/postgresql/postgresql-9.1-main.log -f
	       heading, gifi_accno, contra,
	       tax
	  from account c
	  left join account_link l
	    ON (c.id = l.account_id)
	group by c.id, c.accno, c.description, c.category, c.heading,
	         c.gifi_accno, c.contra, c.tax;
2015-04-20 02:41:29 PDT ERROR:  function public.inventory__get_item_by_partnumber(unknown) does not exist at character 15
2015-04-20 02:41:29 PDT HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
2015-04-20 02:41:29 PDT STATEMENT:  SELECT * FROM "public"."inventory__get_item_by_partnumber"(NULL)
2015-04-20 02:47:39 PDT ERROR:  function public.inventory__get_item_by_partnumber(unknown) does not exist at character 15
2015-04-20 02:47:39 PDT HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
2015-04-20 02:47:39 PDT STATEMENT:  SELECT * FROM "public"."inventory__get_item_by_partnumber"('ENG110')
2015-04-20 03:03:51 PDT ERROR:  duplicate key value violates unique constraint "users_pkey"
2015-04-20 03:03:51 PDT DETAIL:  Key (username)=(kraj) already exists.
2015-04-20 03:03:51 PDT CONTEXT:  SQL statement "insert into users (id, username, entity_id) VALUES (
	                v_user_id,
	                in_username,
	                in_entity_id
	            )"
	PL/pgSQL function "admin__save_user" line 52 at SQL statement
2015-04-20 03:03:51 PDT STATEMENT:  SELECT * FROM "public"."admin__save_user"(NULL, '6', 'kraj', 'xxxxxxx', '1')

**Comment by kraj** [comment]
There were more issues, I tried to update 1.3 migrated user from 'Contacts' menu since this user although a super user 'do-it-all' kind in 1.3 was missing timecard groups under 'user' tab but when I selected them and tried to save the updates to user it ended in below error.

Unable to open script: LedgerSMB::Scripts::employee : No such file or
directory: Can't locate LedgerSMB/Scripts/employee.pm in @INC (@INC
contains: /etc/perl /usr/local/lib/perl/5.14.2
/usr/local/share/perl/5.14.2 /usr/lib/perl5 /usr/share/perl5
/usr/lib/perl/5.14 /usr/share/perl/5.14 /usr/local/lib/site_perl .) at
(eval 1320) line 1.

**Comment by kraj** [comment]
some more logs, before I switched back to old 1.3 based container

2015-04-19 21:25:24 PDT LOG:  received fast shutdown request
2015-04-19 21:25:24 PDT LOG:  aborting any active transactions
2015-04-19 21:25:24 PDT LOG:  autovacuum launcher shutting down
2015-04-19 21:25:24 PDT LOG:  shutting down
2015-04-19 21:25:25 PDT LOG:  database system is shut down
2015-04-20 04:25:48 UTC LOG:  database system was shut down at 2015-04-20 04:25:25 UTC
2015-04-20 04:25:48 UTC LOG:  database system is ready to accept connections
2015-04-20 04:25:48 UTC LOG:  autovacuum launcher started
2015-04-20 04:25:48 UTC LOG:  incomplete startup packet
2015-04-20 04:46:17 UTC FATAL:  password authentication failed for user "legersmb"
2015-04-20 04:46:17 UTC FATAL:  password authentication failed for user "legersmb"
2015-04-20 04:46:21 UTC LOG:  could not receive data from client: Connection reset by peer
2015-04-20 04:46:58 UTC ERROR:  column "customer_id" does not exist
2015-04-20 04:46:58 UTC STATEMENT:  ALTER TABLE partscustomer RENAME customer_id TO credit_id;
2015-04-20 04:46:58 UTC ERROR:  current transaction is aborted, commands ignored until end of transaction block
2015-04-20 04:46:58 UTC STATEMENT:  ALTER TABLE partsvendor RENAME entity_id TO credit_id;
2015-04-20 04:46:59 UTC ERROR:  constraint "entity_bank_account_pkey" of relation "entity_bank_account" does not exist
2015-04-20 04:46:59 UTC STATEMENT:  ALTER TABLE entity_bank_account DROP CONSTRAINT entity_bank_account_pkey;
2015-04-20 04:46:59 UTC ERROR:  current transaction is aborted, commands ignored until end of transaction block
2015-04-20 04:46:59 UTC STATEMENT:  ALTER TABLE entity_bank_account ALTER COLUMN bic DROP NOT NULL;
2015-04-20 04:46:59 UTC ERROR:  current transaction is aborted, commands ignored until end of transaction block
2015-04-20 04:46:59 UTC STATEMENT:  ALTER TABLE entity_bank_account ADD UNIQUE(bic,iban);
2015-04-20 04:46:59 UTC ERROR:  current transaction is aborted, commands ignored until end of transaction block
2015-04-20 04:46:59 UTC STATEMENT:  CREATE UNIQUE INDEX eba_iban_null_bic_u ON entity_bank_account(iban) WHERE bic IS NULL;
2015-04-20 04:46:59 UTC ERROR:  schema "lsmb12" does not exist at character 106
2015-04-20 04:46:59 UTC STATEMENT:  UPDATE parts
           SET income_accno_id = (SELECT account.id
                                    FROM account JOIN lsmb12.chart USING (accno)
                                   WHERE chart.id = income_accno_id),
               expense_accno_id = (SELECT account.id
                                    FROM account JOIN lsmb12.chart USING (accno)
                                   WHERE chart.id = expense_accno_id),
               inventory_accno_id = (SELECT account.id
                                    FROM account JOIN lsmb12.chart USING (accno)
                                   WHERE chart.id = inventory_accno_id)
         WHERE id IN (SELECT id FROM lsmb12.parts op
                       WHERE op.id = parts.id
                             AND (op.income_accno_id = parts.income_accno_id
                                  OR op.inventory_accno_id = parts.inventory_accno_id
                                  or op.expense_accno_id = parts.expense_accno_id));
2015-04-20 04:46:59 UTC ERROR:  column "total" of relation "jcitems" already exists
2015-04-20 04:46:59 UTC STATEMENT:  ALTER TABLE jcitems ADD total numeric NOT NULL DEFAULT 0;
2015-04-20 04:46:59 UTC ERROR:  current transaction is aborted, commands ignored until end of transaction block
2015-04-20 04:46:59 UTC STATEMENT:  ALTER TABLE jcitems ADD non_billable numeric NOT NULL DEFAULT 0;
2015-04-20 04:46:59 UTC ERROR:  current transaction is aborted, commands ignored until end of transaction block
2015-04-20 04:46:59 UTC STATEMENT:  UPDATE jcitems
           SET total = qty
         WHERE qty IS NOT NULL and total = 0;
2015-04-20 04:46:59 UTC ERROR:  column "recon_fx" of relation "cr_report" already exists
2015-04-20 04:46:59 UTC STATEMENT:  ALTER TABLE cr_report ADD recon_fx bool default false;
2015-04-20 04:46:59 UTC ERROR:  column "minvalue" of relation "tax" already exists
2015-04-20 04:46:59 UTC STATEMENT:  ALTER TABLE tax ADD minvalue numeric;
2015-04-20 04:46:59 UTC ERROR:  current transaction is aborted, commands ignored until end of transaction block
2015-04-20 04:46:59 UTC STATEMENT:  ALTER TABLE tax ADD maxvalue numeric;
2015-04-20 04:46:59 UTC ERROR:  column "invoice_include" of relation "mime_type" already exists
2015-04-20 04:46:59 UTC STATEMENT:  ALTER TABLE mime_type ADD invoice_include bool default false;
2015-04-20 04:46:59 UTC ERROR:  current transaction is aborted, commands ignored until end of transaction block
2015-04-20 04:46:59 UTC STATEMENT:  UPDATE mime_type SET invoice_include = 'true' where mime_type like 'image/%';
2015-04-20 04:46:59 UTC ERROR:  relation "ac_transdate_year_idx" already exists
2015-04-20 04:46:59 UTC STATEMENT:  create index ac_transdate_year_idx on acc_trans(EXTRACT ('YEAR' FROM transdate));
2015-04-20 04:46:59 UTC ERROR:  duplicate key value violates unique constraint "batch_class_pkey"
2015-04-20 04:46:59 UTC DETAIL:  Key (class)=(receipt_reversal) already exists.
2015-04-20 04:46:59 UTC STATEMENT:  insert into batch_class (id,class) values (7,'receipt_reversal');

**Comment by kraj** [comment]
apache logs

2015/04/20 04:56:18 - ERROR - LedgerSMB::_error LedgerSMB.pm (650) -- Unable to open script: LedgerSMB::Scripts::employee : No such file or directory: Can't locate LedgerSMB/Scripts/employee.pm in @INC (@INC contains: /etc/perl /usr/local/lib/perl/5.14.2 /usr/local/share/perl/5.14.2 /usr/lib/perl5 /usr/share/perl5 /usr/lib/perl/5.14 /usr/share/perl/5.14 /usr/local/lib/site_perl .) at (eval 1321) line 1.
2015/04/20 04:56:18 - ERROR - LedgerSMB::_error LedgerSMB.pm (651) -- dbversion: 1.4.10, company: burrrr
Died at LedgerSMB.pm line 664.
Compilation failed in require at /var/www/ledgersmb/employee.pl line 7.

**Comment by kraj** [comment]
timecard filling time apache errors

Died at LedgerSMB.pm line 664.
Compilation failed in require at /var/www/ledgersmb/employee.pl line 7.
^[[B^[[B
DBD::Pg::st execute failed: ERROR:  function public.inventory__get_item_by_partnumber(unknown) does not exist
LINE 1: SELECT * FROM "public"."inventory__get_item_by_partnumber"('...
                      ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts. at LedgerSMB.pm line 586.
2015/04/20 04:59:47 - ERROR - LedgerSMB::dberror LedgerSMB.pm (773) -- Logging SQL State 42883, error 7, string ERROR:  function public.inventory__get_item_by_partnumber(unknown) does not exist
LINE 1: SELECT * FROM "public"."inventory__get_item_by_partnumber"('...
                      ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
2015/04/20 04:59:47 - ERROR - LedgerSMB::_error LedgerSMB.pm (650) -- Internal Database Error
More information has been reported in the error logs at LedgerSMB.pm line 776.
2015/04/20 04:59:47 - ERROR - LedgerSMB::_error LedgerSMB.pm (651) -- dbversion: 1.4.10, company: burrr
Died at LedgerSMB.pm line 664.
Compilation failed in require at /var/www/ledgersmb/timecard.pl line 8.

**Comment by ehuelsmann** [comment]
Ok. So, there are multiple issues here:

1. Adding time cards to a project fails
2. Adding a project fails
3. Adding permissions to a user fails

(3) has been fixed and will be released this week with 1.4.12. 

I can't reproduce (2) on 1.4.12-rc1, but I can reproduce (1) in 1.4.12-rc1; I've traced the issue to "sql/modules/Inventory.sql" not being listed in "sql/modules/LOADORDER". This will be addressed and fixed in 1.4.13.

Thanks again for reporting; sorry that it took some time to be addressed!
